### PR TITLE
[google-apps-script] Replaced ambiguous object type with more specific Record types

### PR DESCRIPTION
**Pull request by marekdedic**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.google.com/drive/api/reference/rest/v3
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

Closes #70472

### CODE REVIEW

**Review by jpoehnelt — APPROVED** [review]

**Comment by marekdedic** [comment]
Sorry, also added a test here :/

**Review by jpoehnelt — APPROVED** [review]

**Comment by jpoehnelt** [comment]
Ready to merge

File: types/google-apps-script/apis/drive_v3.d.ts [hunk 1]
```diff
@@ -343,10 +343,29 @@ declare namespace GoogleAppsScript {
                         /**Unsets the values for this field.*/
                         unsetValues: boolean;
                     }
+                    /**Representation of field, which is a typed key-value pair.*/
```
Comment (marekdedic) [inline types/google-apps-script/apis/drive_v3.d.ts:346]: Removed in #71502 because it was unused... Well, now it's used :D

**Review by jpoehnelt — APPROVED** [review]

**Comment by marekdedic** [comment]
Ready to merge

**Pull request merged**